src/main/java/minicraft/item/WateringCanItem.java

```diff
@@ -0,0 +1,165 @@
+	protected static ArrayList<Item> getAllInstances() {
+		ArrayList<Item> items = new ArrayList<>();
+		items.add(new WateringCanItem("Watering Can"));
+		return items;
+	}
+
+	private static final SpriteLinker.LinkedSprite sprite = new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Item, "watering_tin");
+	private static final SpriteLinker.LinkedSprite spriteFilled = new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Item, "watering_tin_filled");
+
+	private static final SpriteLinker.LinkedSprite[] spriteSplash = new SpriteLinker.LinkedSprite[] {
+		new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Entity, "splash_0"),
+		new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Entity, "splash_1"),
+		new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Entity, "splash_2"),
+		new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Entity, "splash_3")
+	};
+
+	public final int CAPACITY = 3600;
+	public int content = 0;
+	private int renderingTick = 0;
+
+	protected WateringCanItem(String name) {
+		super(name, sprite);
+	}
+
+	@Override
+	public boolean interactOn(Tile tile, Level level, int xt, int yt, Player player, Direction attackDir) {
+		if (tile instanceof WaterTile) {
+			content = CAPACITY;
+			updateSprite();
+			return true;
+		} else if (content > 0) {
+			content--;
+			updateSprite();
+			renderingTick++;
+			if (renderingTick == 8) {
+				Random random = new Random();
+				for (int i = 0; i < 4; i++) {
+					SpriteLinker.LinkedSprite splash = spriteSplash[random.nextInt(spriteSplash.length)];
+					// 2-pixel deviation for centering particle sprites.
+					int destX = player.x - 2 + 12 * attackDir.getX() + random.nextInt(9) - 4;
+					int destY = player.y - 2 + 12 * attackDir.getY() + random.nextInt(9) - 4;
+					int x = player.x - 2 + 4 * attackDir.getX() + random.nextInt(5) - 2;
+					int y = player.y - 2 + 4 * attackDir.getY() + random.nextInt(5) - 2;
+					level.add(new WaterParticle(x, y, 80 + random.nextInt(61) - 30, splash, destX, destY));
+					renderingTick = 0;
+				}
+			}
+			if (tile instanceof CropTile) {
+				int fertilization = ((CropTile) tile).getFertilization(level.getData(xt, yt));
+				if (fertilization < 150) { // Maximum of 5 levels watering can can fertilize.
+					((CropTile) tile).fertilize(level, xt, yt, 1);
+				}
+			} else if (tile instanceof DirtTile) {
+				for (Tile t : level.getAreaTiles(xt, yt, 1)) {
+					if (t instanceof GrassTile) { // Grass tile exists.
+						if (new Random().nextInt(10) == 0)
+							level.setTile(xt, yt, Tiles.get("grass")); // Grass extends.
+						break; // Operation finished.
+					}
+				}
+			} else if (tile instanceof GrassTile) {
+				for (Point p : level.getAreaTilePositions(xt, yt, 1)) {
+					Tile t = level.getTile(p.x, p.y);
+					if (t instanceof DirtTile) { // Dirt tile exists.
+						if (new Random().nextInt(15) == 0)
+							level.setTile(p.x, p.y, Tiles.get("grass")); // Grass extends.
+						break; // Operation finished.
+					}
+				}
+			}
```

Comment:
Wouldn't it be better to first get the nine tiles around the player and then loop those and check if you can water them. This current way of doing it will probably just be confusing.

Reply:
As each target tile does different effect when watering, it might not be a better idea.

Reply:
That's exactly why it would be better. The tile you are standing on wouldn't matter, which would let you water neighbouring tiles. Plus it would look cleaner.

Reply:
I have updated it.